docs/tools/testing/testing_greentea.md

```diff
@@ -178,6 +314,191 @@ This detects your attached target and drives the test. If you need to rerun the
 
 For an explanation of the arguments used in this command, please run `mbedhtrun --help`.
 
+### Command-line usage
+
+This section highlights a few of the capabilities of the Greentea command-line interface. For a full list of the available options, please run `mbed test --help`.
+
+#### Listing all tests
+
+You can use the `-l` argument to list all available tests:
+
+```
+$ mbed test -l
+[mbed] Working path "/Users/janjon01/repos/first-greentea-test" (program)
+Test Case:
+    Name: mbed-os-components-storage-blockdevice-component_flashiap-tests-filesystem-fopen
+    Path: ./mbed-os/components/storage/blockdevice/COMPONENT_FLASHIAP/TESTS/filesystem/fopen
+Test Case:
+    Name: mbed-os-features-cellular-tests-api-cellular_device
+    Path: ./mbed-os/features/cellular/TESTS/api/cellular_device
+
+...
+```
+
+#### Executing all tests
+
+The default action of Greentea using `mbed test` is to execute all tests that are found. You can also add `-v` to make the output more verbose.
+
+#### Limiting tests
+
+You can select test cases by name using the `-n` argument. This command executes all tests named `tests-mbedmicro-rtos-mbed-mail` from all builds in the test specification:
+
+```
+$ mbed test -n tests-mbedmicro-rtos-mbed-mail
+```
+
+When using the `-n` argument, you can use the `*` character at the end of a test name to match all tests that share a prefix. This command executes all tests that start with `tests-mbedmicro-rtos`:
+
+```
+$ mbed test -n tests-mbedmicro-rtos-*
+```
+
+You can use a comma (`,`) to separate test names (argument `-n`) and build names (argument `-t`). This command executes the tests `tests-mbedmicro-rtos-mbed-mail` and `tests-mbed_drivers-c_strings` for the `K64F-ARM` and `K64F-GCC_ARM` builds in the test specification:
+
+```
+$ mbed test -n tests-mbedmicro-rtos-mbed-mail,tests-mbed_drivers-c_strings -t K64F-ARM,K64F-GCC_ARM
+```
+
+#### Selecting platforms
+
+You can limit which boards Greentea uses for testing by using the `--use-tids` argument.
+
+```
+$ mbed test --use-tids 02400203C3423E603EBEC3D8,024002031E031E6AE3FFE3D2
+```
+
+Where `02400203C3423E603EBEC3D8` and `024002031E031E6AE3FFE3D` are the target IDs of platforms attached to your system.
+
+You can view target IDs using Mbed CLI:
+
+```
+$ mbed detect
++--------------+---------------------+------------+------------+-------------------------+
+|platform_name |platform_name_unique |mount_point |serial_port |target_id                |
++--------------+---------------------+------------+------------+-------------------------+
+|K64F          |K64F[0]              |E:          |COM160      |024002031E031E6AE3FFE3D2 |
+|K64F          |K64F[1]              |F:          |COM162      |02400203C3423E603EBEC3D8 |
+|LPC1768       |LPC1768[0]           |G:          |COM5        |1010ac87cfc4f23c4c57438d |
++--------------+---------------------+------------+------------+-------------------------+
```

Comment:
If your current working directly is within an Mbed OS project, the output of `mbed detect` looks like this:

```
mbed detect
[mbed] Working path "C:\Users\bridan01\onedrive_arm\Documents\dev\mbed-os" (program)

[mbed] Detected K64F, port COM79, mounted F:, interface version 0250:
[mbed] Supported toolchains for K64F
| Target        | mbed OS 2 | mbed OS 5 | uARM |    IAR    |    ARM    |  GCC_ARM  |
|---------------|-----------|-----------|------|-----------|-----------|-----------|
| K64F          | Supported | Supported |  -   | Supported | Supported | Supported |
| RAPIDIOT_K64F | Supported | Supported |  -   | Supported | Supported | Supported |
Supported targets: 2


[mbed] Detected K64F, port COM78, mounted D:, interface version 0251:
[mbed] Supported toolchains for K64F
| Target        | mbed OS 2 | mbed OS 5 | uARM |    IAR    |    ARM    |  GCC_ARM  |
|---------------|-----------|-----------|------|-----------|-----------|-----------|
| K64F          | Supported | Supported |  -   | Supported | Supported | Supported |
| RAPIDIOT_K64F | Supported | Supported |  -   | Supported | Supported | Supported |
Supported targets: 2
```
Which unfortunately does not list the target id. `mbedls` will but I know we want to direct folks to use Mbed CLI commands. In this case I don't think we have many options here, any ideas @theotherjimmy?

Reply:
Oh, huh. Not only does `mbed detect` omit the information, it also neglects to use unique names. I suppose we'll have to use `mbedls` until we fix this in `mbed detect`.